src/Agent/NewRelic/Agent/Core/DataTransport/DataStreamingService.cs

```diff
@@ -425,10 +549,13 @@ public void Shutdown(bool withRestart)
             _grpcWrapper.Shutdown();
         }
 
-        private IClientStreamWriter<TRequest> GetRequestStreamWithRetry(int consumerId, CancellationToken cancellationToken)
+        private bool GetRequestStreamWithRetry(int consumerId, CancellationToken cancellationToken, out IClientStreamWriter<TRequest> requestStream, out IAsyncStreamReader<TResponse> responseStream)
```

Comment:
Should we just return `AsyncDuplexStreamingCall<TRequest, TResponse>` instead of the two out parameters instead? `AsyncDuplexStreamingCall` implements IDisposable so it may also be useful so that we can call `Dispose` on that object too.